server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -742,7 +742,8 @@ private synchronized IndexService createIndexService(
     }
 
     private EngineConfigFactory getEngineConfigFactory(final IndexSettings idxSettings) {
-        return new EngineConfigFactory(this.pluginsService, idxSettings);
+        final IndexService indexService = indices.get(idxSettings.getUUID());
```

Comment:
When would IndexService be null? If it is should it fail?

Reply:
This is good question - I cannot say exactly, but since `indices` is just map, the `get` may return `null`, better be ready for it.

Reply:
I think my opinion is that we should only create a codec service in one place and it should be here: https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/index/shard/IndexShard.java#L324. We should pass in the CodecService factory there and get rid of the codec service in EngineConfigFactory.

Reply:
Very good point, looking into it now, thank you!

Reply:
@jmazanec15 delayed the CodeService creation up to the moment shard is initialized, `MapperService` is taken directly from there (if available). Sadly, I had to add new method `newEngineConfig` and deprecate the old one in order to keep backward compatibility.

Reply:
Could we do something like this: https://github.com/jmazanec15/OpenSearch/commit/165f50a3bf9fa067f777acd85770608299362534? This way, we can exclusively use CodecServiceFactory to build codec and not change any function interfaces?

Alternatively, I thought about building CodecService directly in newEngineConfig. In this case, we would switch from passing in CodecService directly to passing in CodecServiceConfig. But, I thought this might cause us to create more CodecServices than we need to.

Reply:
@jmazanec15 hm ... to me, the logger service does not belong here, may be if you share use case, we could think about it

Reply:
@reta Logger is one of the constructor arguments to CodecService. I think the CodecServiceConfig will need to have it to build CodecService - similar to MapperService. That way, we can use CodecServiceFactory to build default CodecService.

Reply:
@jmazanec15 I should have looked it :) Thank you, will do the change shortly

Reply:
Done, thank @jmazanec15 !